File: FlickrApp/ViewModels/SearchAttemptViewModel.m
//
//  SearchAttemptViewModel.m
//  FlickrApp
//
//  Created by aSqar on 27.11.2017.
//  Copyright © 2017 Askar Bakirov. All rights reserved.
//

#import "SearchAttemptViewModel.h"
#import "SearchAttempt.h"

@interface SearchAttemptViewModel()

@end

@implementation SearchAttemptViewModel

-(instancetype)initWithSearchAttempt:(SearchAttempt *)searchAttempt
{
    self = [super init];
    if (self == nil)
        return nil;

    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
    dateFormatter.dateFormat = @"dd/MM/yyyy HH:mm";
    self.dateString = [dateFormatter stringFromDate:searchAttempt.dateSearched];
    self.queryString = searchAttempt.searchTerm;
    
    return self;
}

-(instancetype)initWithQuery:(NSString *)queryString
{
    [[RLMRealm defaultRealm] beginWriteTransaction];
    
    SearchAttempt *searchAttempt = [[SearchAttempt alloc] init];
    searchAttempt.searchTerm = queryString;
    searchAttempt.dateSearched = [NSDate date];
    [[RLMRealm defaultRealm] addOrUpdateObject: searchAttempt];
    
    [[RLMRealm defaultRealm] commitWriteTransaction];
    
    return [self initWithSearchAttempt: searchAttempt];
}

@end

File: FlickrApp/ViewModels/SearchAttemptViewModel.swift
//
//  SearchAttemptViewModel.m
//  FlickrApp
//
//  Created by aSqar on 27.11.2017.
//  Copyright © 2017 Askar Bakirov. All rights reserved.
//

import Foundation

class SearchAttemptViewModel : BaseViewModel {

    var queryString:String!
    var dateString:String!

    init(searchAttempt:SearchAttempt!) {
        super.init()

        let dateFormatter:DateFormatter! = DateFormatter()
        dateFormatter.dateFormat = "dd/MM/YYYY HH:mm"
        self.dateString = dateFormatter.string(from: searchAttempt.dateSearched)
        self.queryString = searchAttempt.searchTerm


    }
}
